[WPE][GTK] Gardening API test `TestWebKit` `WebKit.MouseMoveAfterCrash`

obyknovenius opened the pull request:
#### fcea4a1eae73c77d22762a9a46583de86a134781
<pre>
[WPE][GTK] Gardening API test `TestWebKit` `WebKit.MouseMoveAfterCrash`
<a href="https://bugs.webkit.org/show_bug.cgi?id=261818">https://bugs.webkit.org/show_bug.cgi?id=261818</a>

Unreviewed test gardening.

Fixed in 268095@main.

* Tools/TestWebKitAPI/glib/TestExpectations.json:

Canonical link: <a href="https://commits.webkit.org/268263@main">https://commits.webkit.org/268263@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/edcdcacd0b99638fc3f34c5b02ab76e14757c18b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19154 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/19565 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/20162 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21043 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17928 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/19369 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/22841 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/19694 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/19620 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/19375 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19453 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16661 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21920 "Built successfully") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16651 "5 flakes 4 failures") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17446 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/23815 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/17700 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17621 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/21767 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18211 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/15429 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/17331 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/21692 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2337 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18047 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/UIProcess/WebPageProxy.h:
```diff
@@ -73,6 +75,11 @@ class Decoder;
+
+template<> struct AsyncReplyError<WebKit::WebEventType> {
+    static WebKit::WebEventType create() { return WebKit::WebEventType::NoType; }
+};
```
But there is no `NoType` member of `WebKit::WebEventType`?

mcatanzaro replied:
Yeah, I think you either forgot to build this, or perhaps forgot to commit your changes to that file?

I'd suggest doing this in WebEventType.h instead, even if it requires including Connection.h, for extra safety.

Alternatively, perhaps it would be safe to simply change the default value, since MouseDown is not a good default. _Hopefully_ that won't break anything. If adding WebEventType::NoType you'll probably have to deal with a bunch of -Wswitch warnings anyway.

obyknovenius replied:
Oh, I see. It was removed two days ago in https://github.com/WebKit/WebKit/commit/791e55b6738d6cf322148abc8c15fb39a61527cd.
That PR actually fixed the issue I tried to fix with this PR but in a more verbose way. I like my way more since it's more compact but it's probably too late. Should I close this one now?

mcatanzaro replied:
Aha!

Well, you should still land your test expectations change.

obyknovenius replied:
I've edited the PR to only update the `TestExpectations.json`.

mcatanzaro reviewed (commented):
(Please remember to manually close that bug #85066 when this is fixed, since you created a new bug report for this pull request.)

obyknovenius committed:
[WPE][GTK] Gardening API test `TestWebKit` `WebKit.MouseMoveAfterCrash`
https://bugs.webkit.org/show_bug.cgi?id=261818

Unreviewed test gardening.

Fixed in 268095@main.

* Tools/TestWebKitAPI/glib/TestExpectations.json:

Canonical link: https://commits.webkit.org/268263@main
